### PR TITLE
Implement proper logging and logging to file

### LINKED ISSUE #35: Create a log file
Especially if there are panics that we want to debug.

### PR DESCRIPTION
**Pull request by jieyouxu**
stdout logging

- Log INFO-level to stdout.
- Log DEBUG-level to debug log file at `project_dirs.data_dir().join("drg-mod-integration.log")`.
    - This is `$XDG_DATA_HOME/drg-mod-integration/drg-mod-integration.log` or `$HOME/.local/share/drg-mod-integration/drg-mod-integration.log` on Linux, and `{FOLDERID_RoamingAppData}\drg-mod-integration\drg-mod-integration.log` on Windows.
- Migrate `println`s over to use proper logging infrastructure.

Closes #35.

### CODE REVIEW

File: src/splice.rs [hunk 1]
```diff
@@ -977,13 +977,13 @@ pub fn find_hooks<'a, C: std::io::Read + std::io::Seek>(
...
-    /*
+    let mut debug_jumps = Vec::new();
     for (a, bs) in &jumps {
         for b in bs {
-            println!("{}:{} -> {}:{}", a.0.index, a.1, b.0.index, b.1);
+            debug_jumps.push(format!("{}:{} -> {}:{}", a.0.index, a.1, b.0.index, b.1));
         }
     }
-    */
+    trace!(target = "splice_print_jumps", debug_jumps = ?debug_jumps);
```
Comment (trumank) [inline src/splice.rs:986]: These are pretty much meaningless on their own

File: src/main.rs [hunk 2]
```diff
@@ -107,8 +118,79 @@ fn main() -> Result<()> {
...
+    let f = File::create(project_dirs.data_dir().join("drg-mod-integration.log"))?;
+    let writer = BufWriter::new(f);
+    let (log_file_appender, guard) = tracing_appender::non_blocking(writer);
+    let debug_file_log = fmt::layer()
+        .with_writer(log_file_appender)
+        .fmt_fields(NewType(Pretty::default()))
+        .with_ansi(false)
+        .with_filter(filter::filter_fn(|metadata| {
+            *metadata.level() <= Level::DEBUG && metadata.target() == "drg_mod_integration"
+        }));
```
Comment (trumank) [inline src/main.rs:161]: This seems to filter only messages matching exactly `drg_mod_integration`. I think this is what you want:
```rust
        .with_filter(
            tracing_subscriber::filter::Targets::new()
                .with_target("drg_mod_integration", Level::DEBUG),
        );
```

**Review by trumank — COMMENTED** [review]
I think leaving the logs in the user data directory is fine. They're mostly for us so we can request them if needed.

**Comment by jieyouxu** [comment]
Note that there are probably places where there should be more TRACE/DEBUG/INFO logs to help us debug issues, but I haven't gone through everything in this PR.

**Pull request merged**